mixer/adapter/stackdriver/trace/trace.go

```diff
@@ -182,26 +184,48 @@ func buildSpanData(val *tracespan.Instance, parentContext trace.SpanContext, spa
 	}
 
 	spanKind := trace.SpanKindServer
+	parentSpanID := parentContext.SpanID
+	spanID := spanContext.SpanID
 	if val.ClientSpan {
 		spanKind = trace.SpanKindClient
+		// If this is a client span and rewriteClientSpanId is true, deterministically create a new span
+		// ID and rewrite span id to that one. This id should also be used as server span's parent span
+		// id.
+		if val.RewriteClientSpanId {
+			spanID = rewriteSpanID(spanID)
+		}
+	} else if val.RewriteClientSpanId {
+		// If this is a server span and rewriteClientSpanId is true, deterministically create a new span
+		// ID and rewrite parent id to that one, which makes this span attached to the client span as a
+		// child span.
+		parentSpanID = rewriteSpanID(spanID)
```

Comment:
Just noticed this but @bianpengyuan why do we rewrite the span ID of the parent here? Shouldn't we only rewrite the child span ID?

Reply:
ahh I see it is rewriting the parent span ID on the server side which is the child as your comment mentions, that makes sense sorry